js/maze.js

```diff
@@ -0,0 +1,235 @@
+"use strict";
+
+function Maze(size){
+    
+    if(size === 'small'){
+        this.dimension = 15;
+        this.generationSpeed = 100;
+    }
+    if(size === 'medium'){
+        this.dimension = 25;
+        this.generationSpeed = 60;
+    }
+    if(size === 'large'){
+        this.dimension = 45;
+        this.generationSpeed = 40;
+    }
+    this.cellsArray = this.createCells();
+    this.solution = [];
+
+}
+    
+Maze.prototype.createCells = function(){
+    var count = 0;
+
+    var cellsArray = [];
+    
+    for(var i = 0; i < this.dimension; i++){
+        cellsArray[i] = [];
+        for(var j = 0; j < this.dimension; j++){
+            
+            var cell = new MazeCell(i, j);
+            cellsArray[i][j] = cell;
+        }
+    }
+
+    return cellsArray;
+
+}
+    
+Maze.prototype.showMazeField = function(cellsArray){
```

Comment:
Do you think it's necessary to pass cellsArray as parameter to the methods that could achieve it through this. ?